controllers/AdminController.php

```diff
@@ -294,23 +309,40 @@ public function actionInfo($id)
 
     /**
      * Switches to the given user for the rest of the Session.
+     * When no id is given, we switch back to the original admin user
+     * that started the impersonation.
      *
      * @param int $id
      *
      * @return string
      */
-    public function actionSwitch($id)
+    public function actionSwitch($id = null)
     {
-        $user = $this->findModel($id);
+        if(!Yii::$app->getModule('user')->enableImpersonateUser)
+            throw new ForbiddenHttpException(Yii::t('user', 'Impersonate user is disabled in the application configuration'));
+
+        $key = self::ORIGINAL_USER_SESSION_KEY;
```

Comment:
compare
```
run1(thisisaveryloremipsumstringthatdoesnotlookverygoodwhenrepeatedveryoften);
run2(thisisaveryloremipsumstringthatdoesnotlookverygoodwhenrepeatedveryoften);
run3(thisisaveryloremipsumstringthatdoesnotlookverygoodwhenrepeatedveryoften);
```
to
```php
$key = thisisaveryloremipsumstringthatdoesnotlookverygoodwhenrepeatedveryoften;
run1($key);
run2($key);
run3($key);
```

:) but its just a matter of style, if you want i can adjust it to your style.